src/iam-token.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright 2021 IBM Corp. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const axios = require('axios');
+const objectPath = require('object-path');
+const hash = require('object-hash');
+
+
+module.exports = class IamTokenGetter {
+  constructor() {
+    this.s3TokenCache = {};
+    this.log = require('./bunyan-api').createLogger('IamTokenGetter');
+  }
+  async fetchS3Token(iam, kubeResourceMeta, namespace) {
+    let apiKey;
+    const apiKeyAlpha1 = objectPath.get(iam, 'api_key');
+    const apiKeyStr = objectPath.get(iam, 'apiKey');
+    const apiKeyRef = objectPath.get(iam, 'apiKeyRef');
+
+    if (typeof apiKeyAlpha1 == 'string') {
+      apiKey = apiKeyAlpha1;
+    } else if (typeof apiKeyStr == 'string') {
+      apiKey = apiKeyStr;
+    } else if (typeof apiKeyAlpha1 == 'object') {
+      let secretName = objectPath.get(apiKeyAlpha1, 'valueFrom.secretKeyRef.name');
```

Comment:
`const` instead of `let` or `var` wherever possible (here and other places)

Reply:
done